opm/grid/cpgrid/CpGridData.hpp

```diff
@@ -309,6 +309,31 @@ class CpGridData
         ijk[2] = gc / logical_cartesian_size_[1];
     }
 
+    /// @brief Extract in-parent-cell Cartesian index triplet (i,j,k) of a refined cell.
+    ///
+    /// - If refinement has been done via LGRs, there amount of children of each parent cell in each
+    /// direction (x,y,and z) is given by std::array<int,3> cells_per_dim. In this case, CpGrid::adapt(/* with args */)
+    /// has been called.
+    /// - If refinement has been done via adapt() (without arguments) the default amount of children of each parent
+    /// cell in each direction is cells_per_dim = {2,2,2}.
+    ///
+    /// @param [in] idx_in_parent_cell      Integer between 0 and cells_per_dim[0]*cells_per_dim[1]*cells_per_dim[2]-1
+    ///                                     (total amount of children of a parent cell minus one since starting index is zero).
+    /// @param [in] cells_per_dim
+    /// @param [out] ijk                    In-parent-cell Cartesian index triplet
+    void getInParentCellIJK(int idx_in_parent_cell, const std::array<int,3>& cells_per_dim,  std::array<int,3>& ijk) const
```

Comment:
Isn't this much more general than the name and the documentation suggests? It seems to convert a cartesian index to ijk and has no relation to refinement. You just happen to use it during refinement, currently. Why not simply name it getIJK?

In addition we could just return the ijk directly and only have two parameters.
Then you can do this in line 306: `ijk = getIK(c, logical_cartesian_size_);` and that method is done.

Reply:
It sounds doable. I kept both methods since CpGrid::getIJK  uses current_view_data_ ->getIJK (which is CpGridData::getIJK), and the integer argument "c" then it's used to get global_cell_[c] and then with this new integer, the "usual repeated deduction of ijk is done"

Reply:
It's probably your comment combined with ijk = getIJK(global_cell_[c], logical_cartesian_size_)